src/main/java/io/bitrise/trace/step/InjectTraceTask.java

```diff
@@ -254,8 +256,18 @@ static boolean updateBuildScriptContent(final String path) throws IOException {
         final String regex = "buildscript[ \\t\\n\\r]*\\{";
         final Pattern pattern = Pattern.compile(regex);
         final Matcher matcher = pattern.matcher(codeContent);
+        final List<Range> literalsPos = findStringLiterals(codeContent);
         if (matcher.find()) {
-            final String updatedContent = matcher.replaceFirst(getUpdatedBuildScriptContent());
+            boolean result = true;
+            while (literalsPos.stream().anyMatch(it -> it.intersect(matcher.start()))) {
+                result = matcher.find();
+            }
+            if (!result) {
+                return result;
```

Comment:
nit: it'd be nice to have a test that covered when this should return false.

Reply:
Added test case.